File: orchestrator/src/main/java/airportsystem/orchestrator/config/RabbitMQConfig.java
package airportsystem.orchestrator.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;
import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitMQConfig {

    @Bean
    public Queue retornoQueue() {
        return new Queue("retorno", true);
    }

    @Bean public Queue listarClientesQueue() { return new Queue("cliente.listar", true); }

    @Bean
    public ObjectMapper objectMapper() {
        return JsonMapper.builder().findAndAddModules().build();
    }
}

File: saga-orchestrator/src/main/kotlin/com/tads/airport_system/rabbitmq/RabbitMQConfig.kt
package com.tads.airport_system.ms_saga_orchestrator.config

import org.springframework.amqp.core.Queue
import org.springframework.amqp.rabbit.connection.ConnectionFactory
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitMQConfig {

    @Bean
    fun authServiceQueue(): Queue {
        return Queue("auth-service-queue", true)
    }

    @Bean
    fun sagaOrchestratorReplyQueue(): Queue {
        return Queue("saga-orchestrator-reply-queue", true)
    }

    @Bean
    fun jackson2JsonMessageConverter(): Jackson2JsonMessageConverter {
        return Jackson2JsonMessageConverter()
    }

    @Bean
    fun rabbitTemplate(connectionFactory: ConnectionFactory): RabbitTemplate {
        val rabbitTemplate = RabbitTemplate(connectionFactory)
        rabbitTemplate.messageConverter = jackson2JsonMessageConverter()
        return rabbitTemplate
    }
}